tests/test_transitfit.py

```diff
@@ -0,0 +1,296 @@
+"""
+test_transitfit.py - Luke Bouma (luke@astro.princeton.edu) - Aug 2019
+License: MIT - see the LICENSE file for details.
+
+When run, this file downloads two TESS light curves from the
+astrobase-notebooks/nb-data directory.
+
+Implemented tests include:
+
+- fit an easy single-sector light curve (6 big HJ transits) using
+fivetransitparam_fit_magseries, derive uncertainties, check them against
+theoretical predictions.
+
+- fit a trickier single-sector light curve (2 transits), do as above.
+
+- test_multithread_speed: ensure that increasing nworkers speeds up the MCMC
+sampling. 2019/08/15: this test fails.
+"""
+
+###########
+# imports #
+###########
+import os, multiprocessing
+import time as _time
+from glob import glob
+try:
+    from urllib import urlretrieve
+except Exception:
+    from urllib.request import urlretrieve
+
+import numpy as np
+from numpy.testing import assert_allclose
+
+from astropy.io import fits
+import astrobase.imageutils as iu
+
+from astrobase.lcfit.transits import fivetransitparam_fit_magseries
+from test_periodbase import on_download_chunk
+
+##########
+# config #
+##########
+
+# download the light curves used for tests if they do not exist. first is a
+# nice easy hot jupiter, 6 transits.  second is a tricky two-transit warm
+# jupiter. (TOI-450).
+LCURLS = [
+    (
+    "https://github.com/waqasbhatti/astrobase-notebooks/raw/master/nb-data/"
+    "hlsp_cdips_tess_ffi_gaiatwo0003007171311355035136-0006_tess_v01_llc.fits"
+    ),
+    (
+    "https://github.com/waqasbhatti/astrobase-notebooks/raw/master/nb-data/"
+    "hlsp_cdips_tess_ffi_gaiatwo0004827527233363019776-0006_tess_v01_llc.fits"
+    )
+]
+
+modpath = os.path.abspath(__file__)
+LCPATHS = [
+    os.path.abspath(os.path.join(
+        os.getcwd(),
+        'hlsp_cdips_tess_ffi_gaiatwo0003007171311355035136-0006_tess_v01_llc.fits')
+    ),
+    os.path.abspath(os.path.join(
+        os.getcwd(),
+        'hlsp_cdips_tess_ffi_gaiatwo0004827527233363019776-0006_tess_v01_llc.fits')
+    )
+]
+
+for LCPATH, LCURL in zip(LCPATHS, LCURLS):
+    if not os.path.exists(LCPATH):
+        localf, headerr = urlretrieve(
+            LCURL,LCPATH,reporthook=on_download_chunk)
+
+
+#########
+# tests #
+#########
+
+def test_fivetransitparam_fit_magseries_easy():
+    """
+    Fit one TESS sector of data, with a HJ (candidate) in it, for a transit
+    model (t0, period, incl, sma, rp/star) with believable error bars.
+    """
+
+    # path and identifier for GaiaDR2 3007171311355035136
+    lcpath = LCPATHS[0]
+    identifier = str(lcpath.split('gaiatwo')[1].split('-')[0].lstrip('0'))
+
+    hdul = fits.open(lcpath)
```

Comment:
I think it would be good to use the imageutils.get_data_keywords_list here since it's imported above.

Reply:
This was a bit of an ambiguous case to my eye, since the header winds up getting used as well. Regardless -- implemented.

Reply:
done